cub/benchmarks/nvbench_helper/test/gen_entropy.cu

```diff
@@ -0,0 +1,192 @@
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
+
+#include <cub/device/device_run_length_encode.cuh>
+
+#include <thrust/count.h>
+#include <thrust/execution_policy.h>
+#include <thrust/host_vector.h>
+#include <thrust/sort.h>
+
+#include <algorithm>
+#include <array>
+
+#include <catch2/catch.hpp>
+#include <nvbench_helper.cuh>
+
+template <class T>
+double get_expected_entropy(bit_entropy in_entropy)
+{
+  if (in_entropy == bit_entropy::_0_000)
+  {
+    return 0.0;
+  }
+
+  if (in_entropy == bit_entropy::_1_000)
+  {
+    return sizeof(T) * 8;
+  }
+
+  const int samples    = static_cast<int>(in_entropy) + 1;
+  const double p1      = std::pow(0.5, samples);
+  const double p2      = 1 - p1;
+  const double entropy = (-p1 * std::log2(p1)) + (-p2 * std::log2(p2));
+  return sizeof(T) * 8 * entropy;
+}
+
+template <class T>
+double compute_actual_entropy(thrust::device_vector<T> in)
+{
+  const int n = static_cast<int>(in.size());
+
+  #if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
+  thrust::device_vector<T> unique(n);
+  thrust::device_vector<int> counts(n);
+  thrust::device_vector<int> num_runs(1);
+  thrust::sort(in.begin(), in.end(), less_t{});
+
+  // RLE
+  void *d_temp_storage           = nullptr;
+  std::size_t temp_storage_bytes = 0;
+
+  T *d_in             = thrust::raw_pointer_cast(in.data());
+  T *d_unique_out     = thrust::raw_pointer_cast(unique.data());
+  int *d_counts_out   = thrust::raw_pointer_cast(counts.data());
+  int *d_num_runs_out = thrust::raw_pointer_cast(num_runs.data());
+
+  cub::DeviceRunLengthEncode::Encode(d_temp_storage,
+                                     temp_storage_bytes,
+                                     d_in,
+                                     d_unique_out,
+                                     d_counts_out,
+                                     d_num_runs_out,
+                                     n);
+
+  thrust::device_vector<std::uint8_t> temp_storage(temp_storage_bytes);
+  d_temp_storage = thrust::raw_pointer_cast(temp_storage.data());
+
+  cub::DeviceRunLengthEncode::Encode(d_temp_storage,
```

Comment:
question: Wouldn't we need to pre-sort `in` before running `cub::DeviceRunLengthEncode::Encode` here?